### PR TITLE
Added Change Time parameter to File/Directory Rename, added ability to keep source Change Time on File Start Copy

### PR DESCRIPTION
**Pull request by seanmcc-msft**

### CODE REVIEW

File: sdk/storage/Azure.Storage.Files.Shares/src/autorest.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ Run `dotnet build /t:GenerateCode` to generate code.
 
 ``` yaml
 input-file:
-    - https://github.com/Azure/azure-rest-api-specs/blob/f23b5240a97736f1f04bfe49fba5a393903457f1/specification/storage/data-plane/Microsoft.FileStorage/preview/2021-06-08/file.json
+    - C:\Users\seanmcc\git\azure-rest-api-specs\specification\storage\data-plane\Microsoft.FileStorage\preview\2021-06-08\file.json
```
Comment (seanmcc-msft) [inline sdk/storage/Azure.Storage.Files.Shares/src/autorest.md:7]: TODO change this after https://github.com/Azure/azure-rest-api-specs/pull/18054 merges

File: sdk/storage/Azure.Storage.Files.Shares/src/ShareFileClient.cs [hunk 2]
```diff
@@ -1434,6 +1434,12 @@ private void SetNameFieldsIfNull()
                         throw new ArgumentException($"{nameof(ShareFileCopyOptions)}.{nameof(ShareFileCopyOptions.SmbProperties)}.{nameof(ShareFileCopyOptions.SmbProperties.FileLastWrittenOn)} and {nameof(ShareFileCopyOptions)}.{nameof(CopyableFileSmbProperties)}.{nameof(CopyableFileSmbProperties.LastWrittenOn)} cannot both be set.");
                     }
 
+                    if ((copyableFileSmbProperties.GetValueOrDefault() & CopyableFileSmbProperties.ChangedOn) == CopyableFileSmbProperties.ChangedOn
+                        && smbProperties?.FileChangedOn != null)
+                    {
+                        throw new ArgumentException($"{nameof(ShareFileCopyOptions)}.{nameof(ShareFileCopyOptions.SmbProperties)}.{nameof(ShareFileCopyOptions.SmbProperties.FileChangedOn)} and {nameof(ShareFileCopyOptions)}.{nameof(CopyableFileSmbProperties)}.{nameof(CopyableFileSmbProperties.ChangedOn)} cannot both be set.");
```
Comment (amnguye) [inline sdk/storage/Azure.Storage.Files.Shares/src/ShareFileClient.cs:1440]: Can you add this to the Errors.cs?

File: sdk/storage/Azure.Storage.Files.Shares/tests/FileClientTests.cs [hunk 3]
```diff
@@ -1632,6 +1632,33 @@ public async Task StartCopyAsync_CopySourceFileLastWrittenOnError()
                 e.Message));
         }
 
+        [RecordedTest]
+        public async Task StartCopyAsync_CopySourceFileChanagedOnError()
```
Comment (amnguye) [inline sdk/storage/Azure.Storage.Files.Shares/tests/FileClientTests.cs:1636]: NIT Typo: change StartCopyAsync_CopySourceFileChanagedOnError to
StartCopyAsync_CopySourceFileChangedOnError

**Review by amnguye — APPROVED** [review]
LGTM  with just two minor comments

**Pull request merged**